### PR TITLE
Runtime image packaging feature

### PR DESCRIPTION
**Pull request by emersonknapp**
Allow user to optionally package up a docker image containing the resulting build

### CODE REVIEW

File: ros_cross_compile/ros_cross_compile.py [hunk 1]
```diff
@@ -189,10 +198,17 @@ def cross_compile_pipeline(
...
         custom_rosdep_script,
         custom_data_dir,
-        custom_setup_script)
+        custom_setup_script,
+        args.runtime_tag)
+
+    skip = set(args.stages_skip)
+
+    # Only package the runtime image if the user has specified a tag for it
+    if not args.runtime_tag:
```
Comment (jaisontj) [inline ros_cross_compile/ros_cross_compile.py:207]: Since this arg has a default, will this ever be true?

File: ros_cross_compile/runtime.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    NAME = 'runtime'
+
+    def __init__(self):
+        super().__init__(self.NAME)
+
+    def __call__(
+        self, platform: Platform, docker_client: DockerClient, ros_workspace_dir: Path,
+        options: PipelineStageOptions,
+        data_collector: DataCollector
```
Comment (jaisontj) [inline ros_cross_compile/runtime.py:60]: `data_collector` not used?

File: ros_cross_compile/ros_cross_compile.py (hunk 1 above)
Comment (jaisontj) [inline ros_cross_compile/ros_cross_compile.py:207]: Since `arg.runtime_tag` has a default value, will this ever be true?

**Review by jaisontj — COMMENTED** [review]
LGTM except two minor questions

File: ros_cross_compile/ros_cross_compile.py (hunk 1 above)
Comment (emersonknapp) [reply ros_cross_compile/ros_cross_compile.py:207]: Oops, good catch, it was supposed to deafult to None

File: ros_cross_compile/runtime.py (hunk 2 above)
Comment (emersonknapp) [reply ros_cross_compile/runtime.py:60]: oh, good point, I can use it to report the size of the runtime docker image

**Review by dabonnie — CHANGES_REQUESTED** [review]

**Review by dabonnie — APPROVED** [review]

**Pull request merged**